### PR TITLE
Typescript Migration: typeahead

### PR DESCRIPTION
**Pull request by markxiong0122**
migrated chromedash-typeahead and its test file

### CODE REVIEW

File: client-src/elements/chromedash-typeahead.ts [hunk 1]
```diff
@@ -77,35 +75,36 @@ export class ChromedashTypeahead extends LitElement {
...
-    this.slDropdownRef.value.show();
+    this.slDropdownRef.value!.show();
   }
 
   findPrefix() {
-    const inputEl = this.slInputRef.value.input;
+    const inputEl = this.slInputRef.value?.input;
+    if (!inputEl) return;
     const wholeStr = inputEl.value;
     const caret = inputEl.selectionStart;
```
Comment (jrobbins) [inline client-src/elements/chromedash-typeahead.ts:98]: Could we avoid repeatedly using `!` below by changing this to something like this?
`const caret: Number = inputEl.selectionStart || 0;`

File: client-src/elements/chromedash-typeahead.ts [hunk 2]
```diff
@@ -123,37 +122,40 @@ export class ChromedashTypeahead extends LitElement {
 
   async handleCandidateSelected(e) {
     const candidateValue = e.detail.item.value;
-    const inputEl = this.slInputRef.value.input;
-    const wholeStr = inputEl.value;
+    const inputEl = this.slInputRef.value?.renderRoot.querySelector('input');
```
Comment (jrobbins) [inline client-src/elements/chromedash-typeahead.ts:125]: I think we could avoid repeated use of `!` below by bailing out here if `inputEl`is not found.
`if (!inputEl) return;`

File: client-src/elements/chromedash-typeahead_test.ts [hunk 3]
```diff
@@ -199,13 +202,13 @@ describe('chromedash-typeahead-item', () => {
...
     `);
-    const valueEl = component.shadowRoot.querySelector('#value');
-    assert.include(valueEl.innerHTML, 'aVal');
-    assert.include(valueEl.innerHTML, 'ue');
-    const bold = component.shadowRoot.querySelector('b');
-    assert.include(bold.innerHTML, 'aVal');
-    const docEl = component.shadowRoot.querySelector('#doc');
-    assert.include(docEl.innerHTML, 'Docs about it');
+    const valueEl = component.shadowRoot!.querySelector('#value');
+    assert.include(valueEl!.innerHTML, 'aVal');
```
Comment (jrobbins) [inline client-src/elements/chromedash-typeahead_test.ts:206]: Can we avoid the repeated `!`s below by adding:
`assert.exists(valueEl);`?

**Review by jrobbins — APPROVED** [review]
Looks good to me after considering a few comments.

File: client-src/elements/chromedash-typeahead.ts (hunk 2 above)
Comment (markxiong0122) [reply client-src/elements/chromedash-typeahead.ts:125]: sounds good

File: client-src/elements/chromedash-typeahead_test.ts (hunk 3 above)
Comment (markxiong0122) [reply client-src/elements/chromedash-typeahead_test.ts:206]: yeah that's a good idea, rendorRoot is not applicable to those so I used ! in the first place

**Pull request merged**